Fetch shall get all branches

tyyko opened the pull request:
Fix the prepare deployment error (#112)when a specific commit is set on the
deployment manifest by "ref:" and the commit is not part of the default
branch.

This is a small change that allows to fetch not only the default branch but also all the others.

rkoster commented:
@tyyko, thanks for the PR, would it be possible to add a unit test for this? So this usecase is remembered when someone refactors this functionality at a later point in time?

tyyko commented:
@rkoster , I added two tests for my fix. Of course I needed to edit one of the dummy git releases to archive that. This is what I did: 
```
unzip the compressed file named foo-boshrelease-repo.zip in a temp folder
navigate into the folder and run 'git reset --hard HEAD' to take a look at all the files in the repo
go back to Release 9 commit (the latest on master was 12) 'git checkout 3017d'
created a new branch 'git checkout -b hotfix'
added a release file 'touch releases/foo-9.1.yml'
commit 'git commit -m "Release 9.1 (Hotfix)"  '
[same procedure for release 9.2]
go back to master 'git checkout master'

compress only the .git folder 'zip foo-boshrelease-repo.zip .git/ '
```
This change to the test release does not influence the other tests.

rkoster commented:
@tyyko thanks!
